tests/gtest/avif_fuzztest_yuvrgb.cc

```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 Google LLC
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "avif/avif.h"
+#include "avif_fuzztest_helpers.h"
+#include "aviftest_helpers.h"
+#include "avifutil.h"
+#include "fuzztest/fuzztest.h"
+#include "gtest/gtest.h"
+
+using ::fuzztest::Arbitrary;
+using ::fuzztest::ElementOf;
+
+namespace avif {
+namespace testutil {
+namespace {
+
+void Convert(ImagePtr image, uint32_t rgb_depth, avifRGBFormat rgb_format,
+             avifChromaUpsampling upsampling,
+             avifChromaDownsampling downsampling, bool avoid_libyuv,
+             bool ignore_alpha, bool alpha_premultiplied, bool is_float,
+             int max_threads) {
+  AvifRgbImage rgb(image.get(), rgb_depth, rgb_format);
+  rgb.chromaUpsampling = upsampling;
+  rgb.chromaDownsampling = downsampling;
+  rgb.avoidLibYUV = avoid_libyuv;
+  rgb.ignoreAlpha = ignore_alpha;
+  rgb.alphaPremultiplied = alpha_premultiplied;
+  rgb.isFloat = is_float;
+  rgb.maxThreads = max_threads;
+
+  // See avifRGBFormat and avifRGBImage.
+  const avifResult expected_yuv_to_rgb_result =
+      (rgb_format == AVIF_RGB_FORMAT_RGB_565 && rgb_depth != 8)
+          ? AVIF_RESULT_REFORMAT_FAILED
+      : (is_float && rgb_depth != 16) ? AVIF_RESULT_REFORMAT_FAILED
+      : (max_threads < 0)             ? AVIF_RESULT_REFORMAT_FAILED
+                                      : AVIF_RESULT_OK;
+  const avifResult expected_rgb_to_yuv_result =
+      (rgb_format == AVIF_RGB_FORMAT_RGB_565) ? AVIF_RESULT_REFORMAT_FAILED
+      : (is_float)                            ? AVIF_RESULT_NOT_IMPLEMENTED
+                                              : AVIF_RESULT_OK;
+
+  ASSERT_EQ(avifImageYUVToRGB(image.get(), &rgb), expected_yuv_to_rgb_result);
+  ASSERT_EQ(avifImageRGBToYUV(image.get(), &rgb), expected_rgb_to_yuv_result);
+}
+
+FUZZ_TEST(YuvRgbFuzzTest, Convert)
+    .WithDomains(ArbitraryAvifImage(),
+                 /*rgb_depth=*/ElementOf({8, 10, 12, 16}),
+                 ElementOf({AVIF_RGB_FORMAT_RGB, AVIF_RGB_FORMAT_RGBA,
```

Comment:
Could be better to use: `InRange(0, AVIF_RGB_FORMAT_COUNT)` (`AVIF_RGB_FORMAT_COUNT` is also an invalid value which could be useful to test against)